src/model/cloud-runner/error/cloud-runner-error.ts

```diff
@@ -1,16 +1,16 @@
 import CloudRunnerLogger from '../services/cloud-runner-logger';
-import { CloudRunnerState } from '../state/cloud-runner-state';
 import * as core from '@actions/core';
+import CloudRunner from '../cloud-runner';
 
 export class CloudRunnerError {
   public static async handleException(error: unknown) {
     CloudRunnerLogger.error(JSON.stringify(error, undefined, 4));
     core.setFailed('Cloud Runner failed');
-    await CloudRunnerState.CloudRunnerProviderPlatform.cleanupSharedResources(
-      CloudRunnerState.buildParams.buildGuid,
-      CloudRunnerState.buildParams,
-      CloudRunnerState.branchName,
-      CloudRunnerState.defaultSecrets,
+    await CloudRunner.CloudRunnerProviderPlatform.cleanupSharedResources(
```

Comment:
Perhaps we could rename `CloudRunner.CloudRunnerProviderPlatform` to `CloudRunner.Provider`.

We could go one step further and encapsulate knowledge about the Interface to CloudRunner itself. You can do this by calling `CloudRunner.cleanup()` instead. It would then call `CloudRunner.Provider.cleanup()` which would then know how to clean up it's shared resources.